teos/src/api/internal.rs

```diff
@@ -566,6 +610,30 @@ mod tests_private_api {
                     )) if Locator::from_slice(app_loc).unwrap() == locator
                 ));
             }
+
+            // Query for the current locator with the optional user_id present
+            let response = internal_api
+                .get_appointments(Request::new(msgs::GetAppointmentsRequest {
+                    locator: locator.to_vec(),
+                    user_id: Some(random_user_id.to_vec()),
+                }))
+                .await
+                .unwrap()
+                .into_inner();
+
+            // Verify that only a single appointment is returned
+            assert_eq!(response.appointments.len(), 1 );
```

Comment:
How is it supposed to be only one appointment? Shouldn't it be `random_appointment_num`?

Reply:
Aha, since we use the same `dispute_txid` all the `random_appointment_num` (`our_appointment_count`) overwrite each others. I think that's another good reason to move out testing this to another test where we can use different `dispute_txid`s.